### ISSUE TITLE
Errors format is not compatible with JSONAPI

### ISSUE THREAD

**Issue by yairpe** [tuyakhov/yii2-json-api#29]
Thank you for this extension. I'm starting to use it and found some incomptability with the jsonapi 1.0 standard.

Here is how a sample error looks like with the extension:
```
{
    "errors": [
        {
            "name": "Not Found",
            "message": "Page not found.",
            "code": 0,
            "status": 404,
            "type": "yii\\web\\NotFoundHttpException",
            "previous": {
                "name": "Invalid Route",
                "message": "Unable to resolve the request \"contries/index\".",
                "code": 0,
                "type": "yii\\base\\InvalidRouteException"
            }
        }
    ]
}
```

This is because the API response formatter just put the data Yii generated data inside the error section

`$apiDocument = ['errors' => $response->data];`

A [JSONAPI error](http://jsonapi.org/format/1.0/#errors) has a different structure 

What is missing IMHO is a mapping between Yii way of representing errors to JSONAPI compatible format. 

For example if we have an error like above, the formatter should convert it to something like
```
{
  "errors": [
    {
      "status": "404",
      "title":  "Resource Not Found",
      "detail": Resource not found. Caused by Invalid Route. Unable to resolve the request "contries/index"
    }
  ]
}
```

**Comment by acorncom** [comment]
From the way I read the spec on errors, all discussion of errors state that errors MAY have the following attributes, not that they MUST. 

That said, having title and detail would be quite handy ... :grinning:

**Comment by mtangoo** [comment]
+1 for the latter format. It is a way cleaner than the former. @yairpe do you have time to make PR for that?

**Comment by tuyakhov** [comment]
Resolved in #42
